module/x/gravity/keeper/ethereum_event_handler.go

```diff
@@ -82,7 +86,39 @@ func (a EthereumEventProcessor) Handle(ctx sdk.Context, eve types.EthereumEvent)
 		})
 		a.keeper.AfterSignerSetExecutedEvent(ctx, *event)
 		return nil
+	case *types.SendToIBCEvent:
+		// Check if coin is Cosmos-originated asset and get denom
+		isCosmosOriginated, denom := a.keeper.ERC20ToDenomLookup(ctx, event.TokenContract)
+		addr := sdk.AccAddress(event.CosmosReceiver)
+		coin := sdk.NewCoin(denom, event.Amount)
+		coins := sdk.Coins{coin}
+
+		if !isCosmosOriginated {
+			if err := a.DetectMaliciousSupply(ctx, denom, event.Amount); err != nil {
+				return err
+			}
+
+			// if it is not cosmos originated, mint the coins (aka vouchers)
+			if err := a.bankKeeper.MintCoins(ctx, types.ModuleName, coins); err != nil {
+				return sdkerrors.Wrapf(err, "mint vouchers coins: %s", coins)
+			}
+		}
+
+		if err := a.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, addr, coins); err != nil {
+			return err
+		}
 
+		// format of this could be {port}/{channel}
+		chanport := strings.Split(event.Channel, "/")
+
+		// TODO: here we use addr.String() we need to be able to use the acc address here encoded in the recieving chain
+		// bech32 format. We can get around this in 2 ways: one is sending it over from ETH, other is figuring out how to encode
+		// the bytes with a custom prefix which seems like it should be easy.
+		if err := a.transferKeeper.SendTransfer(ctx, chanport[0], chanport[1], coin, addr, addr.String(), clienttypes.NewHeight(0, 0), uint64(time.Second*100)); err != nil {
+			return err
```

Comment:
On this error we my want to default to just dropping the tokens in the users wallet on the gravity chain

Reply:
(i.e. in the event that channel or port doesn't exist)